Fix for attachments in Svelte useChat

tyfiero opened the pull request:
Made a fix for attachments to work with Svelte. Took me longer than I'd like to admit to find out how to fix this. Works well for us

cs50victor commented:
🚀. this was a very tricky bug to figure out.

lgrammel commented:
cc @jeremyphilemon - was this already implemented or would the solution be more involved (with file lists etc)? I don't want Svelte to have deviate from what we implemented in React

tyfiero commented:
> cc @jeremyphilemon - was this already implemented or would the solution be more involved (with file lists etc)? I don't want Svelte to have deviate from what we implemented in React

Well this does work with multiple files in experimental_attachments we've tested it, its essentially the same code from the react package, I just added the two lines that were missing in svelte.

lgrammel commented:
@tyfiero the changes for attachment support in React were more complex:
- https://github.com/vercel/ai/pull/2226
- https://github.com/vercel/ai/pull/2732
- https://github.com/vercel/ai/pull/2976

is all of this implemented for Svelte?

tyfiero commented:
We opted to port our app over to react from svelte. It's clear the svelte package doesn't have, and will never have, the same support that the react package will. I won't have the time to implement those other attachment features in svelte. This PR can be closed.

MentalGear commented:
I think feature parity should be an overall goal, no matter the front end library - especially for **svelte** that is officially supported by vercel.

jstjoe commented:
Hey, are there any known plans to implement this functionality @lgrammel ? I'm also hoping to see this feature come to Svelte and just want to plan accordingly. Thank you!
